### PR TITLE
[WIP] Support for java.nio.file.SystemWatcher for file/directory watching events

### PR DESCRIPTION
**Pull request by hmemcpy**
I started playing around with Monix today and got inspired to try to wrap the System Watcher.

I've added a helper method `watchAsync` that takes in a `Path` and optional arguments to specify what events to watch for. The observable receives an `Array[WatchEvent[_]]`.

Here's the simplest use case:

```scala
implicit val ctx = monix.execution.Scheduler.Implicits.global

val path = Paths.get("/tmp")

def printEvent(event: WatchEvent[_]): Unit = {
  val name = event.context().toString
  val fullPath = path.resolve(name)
  println(s"${event.kind().name()} - $fullPath")
}

watchAsync(path)
  .foreach(p => p.foreach(printEvent))
```

Will output (for a newly created, then deleted file):

```
ENTRY_CREATE - /tmp/hello-world.txt
ENTRY_DELETE - /tmp/hello-world.txt
```

Obviously it borrows heavily from an existing implementation of AsyncChannel. Any comments are welcomed!
Still need to create proper test cases.

### CODE REVIEW

File: src/main/scala/monix/nio/WatchService.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package monix.nio
```
Comment (radusw) [inline src/main/scala/monix/nio/WatchService.scala]: If the service is used only for files we should move it into the file package

File: src/main/scala/monix/nio/WatchServiceObservable.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package monix.nio
```
Comment (radusw) [inline src/main/scala/monix/nio/WatchServiceObservable.scala]: If the service is used only for files we should move it into the file package. 
Also, please check if we can reuse some of the logic

**Comment by hmemcpy** [comment]
Whoa! Thanks @radusw for mentioning me in the sbt issue :)
I'll address the issues soon, will add some more tests as well!

**Comment by radusw** [comment]
Also, please update the README with an example

**Pull request merged**